experimental/local_file_system_memory/local_filesystem_memory.py

```diff
@@ -0,0 +1,260 @@
+"""LocalMemoryAgent abstraction."""
+import os
+import json
+from llama_index import GPTSimpleVectorIndex, GPTListIndex
+from llama_index import LLMPredictor, Document
+from langchain.chat_models import ChatOpenAI
+import logging
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+class LocalFilesystemMemory:
+    """
+    LocalFilesystemMemory is a layer over Llama Index's GPTSimpleVectorIndex which
+    abstracts away the details of managing indexes saved on disk. It assumes /
+    creates a directory structure for storing indexes and provides a simple /
+    interface for managing them.
+    The directory structure is as follows:
+    - index_folder/
+        -  metadata.json
+        -  indexes/
+            -  index_1.json
+            -  index_2.json
+            -  ...
+    The metadata.json file contains the following information:
+    {
+        "index_count": 2,
+        "index_descriptions": [
+            {
+                "name": "index_1",
+                "description": "Description of index generated by GPT Index.",
+            },
+            {
+                "name": "index_2",
+                "description": "Description of index generated by GPT Index.",
+            },
+        ]
+    }
+    """
+
+    """Initialization and related private methods."""
+
+    def __init__(self, folder, model="gpt-3.5-turbo"):
+        self.folder = folder
+        self.model = model
+        self.index_folder = os.path.join(self.folder, "indexes")
+        self.metadata_file = os.path.join(self.folder, "metadata.json")
+        self.__create_folder_structure_if_not_exists()
+
+        # Load the indexes from disk if they exist.
+        self.indexes = {}
+        self.__load_from_folder_if_exists()
+
+    def __load_from_folder_if_exists(self):
```

Comment:
in general this function, `__create_folder_structure_if_not_exists` could probably be standalone functions instead of part of the class. makes function signatures (what arguments are being passed in) a bit cleaner

Reply:
Yep!